teps/0001-tekton-enhancement-proposal-process.md

```diff
@@ -0,0 +1,369 @@
+
+- feature, and effort tracking document
+- a product requirements document
+- design document
+
+into one file which is created incrementally in collaboration with one
+or more Working Groups (WGs).
+
+## Motivation
+
+For cross project or new project proposal, an abstraction beyond a
+single GitHub issue seems to be required in order to understand and
+communicate upcoming changes to the Tekton community.
+
+In a blog post describing the [road to Go 2][], Russ Cox explains
+
+> that it is difficult but essential to describe the significance of a
+> problem in a way that someone working in a different environment can
+> understand
+
+As a project, it is vital to be able to track the chain of custody for
+a proposed enhancement from conception through implementation.
+
+Without a standardized mechanism for describing important enhancements,
+our talented technical writers and product managers struggle to weave
+a coherent narrative explaining why a particular release is
+important. Additionally, for critical infrastructure such as Tekton,
+adopters need a forward looking road map in order to plan their
+adoption strategy.
+
+Before this proposal, there is not a standard way or template
+to create project enhancements. We rely on
+documents hosted on Google docs, without a standard template explaining
+the change. Once a proposal is done, via a design docs, it tends
+to be hard to follow what happens with the proposal: updates on the proposal in
+reaction to comments, state of the proposal (when is it accepted, or
+rejected).
+
+The purpose of the TEP process is to reduce the amount of "tribal
+knowledge" in our community. By moving decisions from a smattering of
+mailing lists, video calls and hallway conversations into a well
+tracked artifact, this process aims to enhance communication and
+discoverability.
+
+A TEP is broken into sections which can be merged into source control
+incrementally in order to support an iterative development process. An
+important goal of the TEP process is ensuring that the process for
+submitting the content contained in [design proposals][] is both clear
+and efficient. The TEP process is intended to create high quality
+uniform design and implementation documents for WGs to deliberate.
+
+[road to Go 2]: https://blog.golang.org/toward-go2) 
+[design proposals]: https://github.com/kubernetes/community/tree/master/contributors/design-proposals
+
+## Stewardship
+The following DACI model indentifies the responsible parties for TEPs:
+
+**Workstream** | **Driver** | **Approver** | **Contributor** |
+**Informed** --- | --- | --- | --- | ---
+| TEP Process Stewardship | TG members | TG members | TG members | Community |
+| Enhancement delivery | Enhancement Owner | SIG leadership (SIG Chairs + TLs) | Enhancement Implementer(s) (may overlap with Driver) | Community |
+
+*TG members: Tekton Governing board members*
+
+## Reference-level explanation
+
+### What type of work should be tracked by a TEP
```

Comment:
It makes sense, but it feels like a lot of PRs would fit this.
Perhaps we could include a list of examples from past PRs of changes that would/would not qualify for a TEP?

Reply:
Adding some in the doc :angel: :+1: